_search-plugins/knn/filter-search-knn.md

```diff
@@ -1,126 +1,140 @@
+- The number of documents in an index, where one OpenSearch document corresponds to one k-NN vector.
+- The percentage of documents left in the results after filtering. This value depends on the restrictiveness of the filter that you are providing in the query. The most restrictive filter in the table returns 2.5% of documents in the index, while the least restrictive filter returns 80% of documents.
+- The desired number of returned results (k). 
 
-### Filtered search performance
+Once you've estimated the number of documents in your index, the restrictiveness of your filter, and the desired number of nearest neighbors, use the following table to choose a filtering method that optimizes for recall or latency.
 
-Filtering that is tightly integrated with the Lucene HNSW algorithm implementation allows you to apply k-NN searches more efficiently, both in terms of relevancy of search results and performance. Consider, for example, an exact search using post-filtering on a large dataset that returns results slowly and does not ensure the required number of results specified by `k`.
-With this new capability, you can create an approximate k-NN search, apply filters, and get the number of results that you need. To learn more about approximate searches, see [Approximate k-NN search]({{site.url}}{{site.baseurl}}/search-plugins/knn/approximate-knn/).
-
-The HNSW algorithm decides which type of filtering to apply to a search based on the volume of documents and number of `k` points in the index that you search with a filter.
-
-![How the algorithm evaluates a doc set]({{site.url}}{{site.baseurl}}/images/hsnw-algorithm.png)
-
-| Variable | Description |
--- | -- | -- |
-N | The number of documents in the index.
-P | The number of documents in the search set after the filter is applied using the formula P <= N.
-q | The search vector.
-k | The maximum number of vectors to return in the response.
-
-To learn more about k-NN performance tuning, see [Performance tuning]({{site.url}}{{site.baseurl}}/search-plugins/knn/performance-tuning/).
-
-## Filter approaches by use case
-
-Depending on the dataset that you are searching, you might choose a different approach to minimize recall or latency. You can create filters that are:
-
-* Very restrictive: Returns the lowest number of documents (for example, 2.5%).
-* Somewhat restrictive: Returns some documents (for example, 38%).
-* Not very restrictive: Returns the highest number of documents (for example, 80%).
-
-The restrictive percentage indicates the number of documents the filter returns for any given document set in an index.
-
-Number of Vectors | Filter Restrictive Percentage | k | Recall | Latency
--- | -- | -- | -- | --
+Number of documents in an index | Percentage of documents the filter returns | k | Filtering method to use for higher recall | Filtering method to use for lower latency
```

Comment:
Jeff has got everyone going on this program: add pipes before and after all table elements (in case we ever switch to a host that can only interpret a markdown table with this format. Not sure if you think that's worthwhile.